src/dlsproto/client/internal/SharedResources.d

```diff
@@ -247,5 +280,135 @@ public final class SharedResources
 
             return fiber;
         }
+
+        /**********************************************************************
+
+            Gets a single-shot timer.
+
+            Params:
+                period_ms = timer interval in milliseconds
+                timer_dg = delegate to call when timer fires.
+
+            Returs:
+                ITimer interface to a timer to use during the request's lifetime.
+
+        **********************************************************************/
+
+        public ITimer getTimer ( uint period_ms, void delegate ( ) timer_dg )
+        {
+            auto timer = this.acquired_timers.acquire(new Timer);
+            timer.initialise(period_ms, timer_dg);
+            return timer;
+        }
+
+        /**********************************************************************
+
+            Inteface to a timer to be used during the request's lifetime.
+
+        **********************************************************************/
+
+        interface ITimer
+        {
+            /// Starts the timer
+            void start ( );
+
+            /// Cancels the scheduled timer
+            void cancel ( );
+        }
+    }
+
+    /// TODO: perhaps move to swarm and use it here and
```

Comment:
Let's keep it here for now. Proper documentation required :)

Reply:
Added the docs.